pkg/operator/apiserver/audit/audit_policies_test.go

```diff
@@ -81,17 +86,66 @@ func TestGetAuditPolicy(t *testing.T) {
 					},
 				},
 			},
-			goldenFile: "oauth.yaml",
+			goldenFile:  "oauth.yaml",
+			errContains: "",
+		},
+		{
+			name: "multipleCustomRules",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "system:authenticated:oauth",
+						Profile: "WriteRequestBodies",
+					},
+					{
+						Group:   "system:authenticated",
+						Profile: "AllRequestBodies",
+					},
+				},
+			},
+			goldenFile:  "multipleCr.yaml",
+			errContains: "",
+		},
+		{
+			name: "unknownProfile",
+			config: configv1.Audit{
+				Profile: "InvalidString",
+			},
+			errContains: "unknown audit profile \"InvalidString\"",
+		},
+		{
+			name: "unknownCustomRulesProfile",
+			config: configv1.Audit{
+				Profile: "None",
+				CustomRules: []configv1.AuditCustomRule{
+					{
+						Group:   "InvalidGroup",
+						Profile: "InvalidProfile",
+					},
+				},
+			},
+			errContains: "unknown audit profile \"InvalidProfile\" in customRules for group \"InvalidGroup\"",
 		},
-		// TODO: add test with multiple customRules
 	}
 	for _, scenario := range scenarios {
 		t.Run(scenario.name, func(t *testing.T) {
 			// act
 			policy, err := GetAuditPolicy(scenario.config)
-			if err != nil {
+			if len(scenario.errContains) == 0 && err != nil {
+				t.Errorf("Expected no error yet received error")
```

Comment:
nit: `t.Fatalf("Expected no error yet received error: %v", err)` is more typical.

Reply:
fixed

Reply:
Resolved